lib/thumbs/pull_request_worker.rb

```diff
@@ -187,45 +204,62 @@ def try_run_build_step(name, command)
           status[:output] = sanitize_text(output)
           status[:exit_code] = exit_code.to_i
 
-          @build_status[:steps][name.to_sym]=status
+          build_status[:steps][name.to_sym] = status
           debug_message "[ #{name.upcase} ] [#{result.upcase}] \"#{command}\""
           return status
         end
 
       rescue Timeout::Error => e
-        status[:ended_at]=DateTime.now
+        status[:ended_at] = DateTime.now
         status[:result] = :error
-        status[:message] = "Timeout reached (#{@timeout} seconds)"
+        status[:message] = "Timeout reached (#{config_timeout} seconds)"
         status[:output] = e
 
-        @build_status[:steps][name.to_sym]=status
+        build_status[:steps][name.to_sym] = status
         debug_message "[ #{name.upcase} ] [ERROR] \"#{command}\" #{e}"
         return status
       end
     end
 
     def events
-      client.repository_events(@repo).collect { |e| e.to_h }
+      @events ||= octokit_client.repository_events(repo).collect { |e| e.to_h }
     end
 
-    def push_time_stamp(sha)
-      time_stamp=events.collect { |e| e[:created_at] if e[:type] == 'PushEvent' && e[:payload][:head] == sha }.compact.first
-      time_stamp ? time_stamp : pr.created_at
+    def push_timestamp(sha)
+      push_event = events.detect { |e| e[:payload][:head] == sha &&
```

Comment:
The use of `detect` can be reasoned about to be equivalent to `.collect {...}.first`. `detect` returns the first item that matches the _predicate_ block (block returns something truthy (not `nil` or `false`).